### PR TITLE
Update main for v0.6.1 Release

### PR DESCRIPTION
**Pull request by riedgar-ms**
Required updates to `main` for the `v0.6.1` release:

- Bump version in `__init__.py`
- Update links on the static landing page

Don't need to update the doc configuration this time (regex is flexible enough)

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -34,4 +34,4 @@ pypandoc
 sphinx
 sphinx-gallery
 git+https://github.com/Holzhaus/sphinx-multiversion.git
-pydata-sphinx-theme>=0.5.0
\ No newline at end of file
+pydata-sphinx-theme==0.5.2
```
Comment (adrinjalali) [inline requirements-dev.txt:37]: ideally you'd fix this on `main`, in a separate PR, and then `cherry-pick` on the release branch.

File: fairlearn/__init__.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 from .show_versions import show_versions  # noqa: F401
 
 __name__ = "fairlearn"
-__version__ = "0.6.1.dev0"
+__version__ = "0.6.2.dev0"
```
Comment (adrinjalali) [inline fairlearn/__init__.py:12]: This whole PR seems to need to be cherry picked for the release branch, and therefor this change needs to be in a separate PR I think.

File: requirements-dev.txt (hunk 1 above)
Comment (riedgar-ms) [reply requirements-dev.txt:37]: This specific fix is now in #740 , which I'll cherry pick to `release/v0.6.1`. Technically this part doesn't need to go into the release branch, since documentation is always built from `main` (with the different versions due to `sphinx-multiversion` doing its thing.

I also plan on opening an issue to update to `pydata-sphinx-theme` v0.6 (how dare they have the same version as our current release! ;-) ).

File: fairlearn/__init__.py (hunk 2 above)
Comment (riedgar-ms) [reply fairlearn/__init__.py:12]: This fix and the changes to `index.html` need to be on `main`, since the docs are only build from `main`. The subsequent PR (#740 ) will get cherry picked, since otherwise the release breaks.

Comment (adrinjalali) [reply fairlearn/__init__.py:12]: hmm, ok.

**Review by adrinjalali — APPROVED** [review]

File: fairlearn/__init__.py (hunk 2 above)
Comment (riedgar-ms) [reply fairlearn/__init__.py:12]: I'd be open to improved ways of dealing with the versioned docs. I do take your point that this update looks slightly odd.
